### PR TITLE
Implement material iterative prestressing

### PR DESCRIPTION
**Pull request by amgebauer**
This adds a general material wrapper to prestress any material with iterative prestressing.

In collaboration with @datzj and our student Lena Heiß.

### CODE REVIEW

File: src/mat/4C_mat_iterative_prestress.cpp [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+      const Core::LinAlg::Matrix<3, 3>& prestretch_tensor)
+  {
+    Core::LinAlg::Matrix<3, 3> GS_tilde(true);
+    Core::LinAlg::Matrix<3, 3> GS_tildeGT(true);
+    Core::LinAlg::Matrix<3, 3> S_elastic(true);
+
+    // convert original stress to stress like Voigt notation
+    Core::LinAlg::Voigt::Stresses::vector_to_matrix(stress_elastic, S_elastic);
+
+    // comute G*S_tilde*GT
```
Comment (datzj) [inline src/mat/4C_mat_iterative_prestress.cpp:119]:
```suggestion
    // compute G*S_tilde*GT
```

File: src/mat/4C_mat_iterative_prestress.cpp [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+    GS_tilde.multiply_nn(prestretch_tensor, S_elastic);
+    GS_tildeGT.multiply_nt(GS_tilde, prestretch_tensor);
+
+    // comute pk2 in reference configuration
```
Comment (datzj) [inline src/mat/4C_mat_iterative_prestress.cpp:123]:
```suggestion
    // compute pk2 in reference configuration
```

File: src/inpar/4C_inpar_validmaterials.cpp [hunk 3]
```diff
@@ -3596,6 +3596,19 @@ std::shared_ptr<std::vector<std::shared_ptr<Mat::MaterialDefinition>>> Input::va
...
 
+  /*----------------------------------------------------------------------*/
+  // General material wrapper enabling iterative prestressing
+  {
+    auto m = std::make_shared<Mat::MaterialDefinition>("MAT_IterativePrestress",
+        "General material wrapper enabling iterative pretressing for any material",
+        Core::Materials::m_iterative_prestress);
+
+    add_named_int(m, "MATID", "Id of the material");
+    add_named_bool(m, "ACTIVE", "True if prestress should be updated");
```
Comment (datzj) [inline src/inpar/4C_inpar_validmaterials.cpp:3607]: I don't know if the parameter description is fully clear.

File: src/mat/4C_mat_iterative_prestress.hpp [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+    PAR::IterativePrestressMaterial* params_ = nullptr;
+
+    std::vector<Core::LinAlg::Matrix<3, 3>> prestretch_tensor_ = {};
+  };
+}  // namespace Mat
+
+FOUR_C_NAMESPACE_CLOSE
+
+#endif  // !BACI_MAT_ITERATIVE_PRESTRESS_HPP
```
Comment (datzj) [inline src/mat/4C_mat_iterative_prestress.hpp:125]:
```suggestion
#endif  // !FOUR_C_MAT_ITERATIVE_PRESTRESS_HPP
```

**Review by datzj — DISMISSED** [review]
Thank you for preparing this feature to be merged! I only added a few minor comments/typo fixes. I think the functionality was tested thoroughly in the term paper and also in the test you added.

**Comment by amgebauer** [comment]
@datzj Thanks for spotting the typos. I fixed them.

**Review by datzj — DISMISSED** [review]
LGTM, thanks!

**Review by lauraengelhardt — COMMENTED** [review]
I left two suggestions/questions 😊 Otherwise everything looks good!

**Comment by amgebauer** [comment]
Thanks for your review @lauraengelhardt. I added it to the mat_service. On the way, I fixed an error in the pull_back operation and reused it in another material that had its own implementation.

Now ready for a second round of reviews.

**Review by lauraengelhardt — APPROVED** [review]
Look's good, thanks for incorporating the suggestions!

**Review by datzj — APPROVED** [review]
Looks good!

**Pull request merged**